src/compile/ast.rs

```diff
@@ -220,6 +223,9 @@ impl Type {
             Type::Void => {
                 panic!("tried to get default value for void type");
             }
+            Type::Buffer => {
+                panic!("tried to get default value for buffer type");
```

Comment:
I think an empty buffer would be a reasonable default value for a buffer type, is there a reason against using this?